backend/api/src/complete-quest.ts

```diff
@@ -0,0 +1,19 @@
+import { APIError, endpoint, validate } from 'api/helpers'
+import { getUser } from 'shared/utils'
+import { z } from 'zod'
+import { QUEST_TYPES } from 'common/quest'
+import { completeQuestInternal } from 'shared/quest'
+
+const bodySchema = z.object({
+  // must be of type QuestType
+  questType: z.enum(QUEST_TYPES),
+  userId: z.string(),
+})
+
+export const completequest = endpoint(async (req) => {
```

Comment:
It's easier to use `authEndpoint`, which would include the userId of the authenticated user.

Reply:
Then I'd have to get the user every time I want to call completeQuest, which is excessive imo. Anywhere you want to click a share link you have to add the user to the call.

Reply:
Not sure I understand. AuthEndpoint gets the user for you based on who is signed in (second param is auth, user id is `auth.uid`). You don't have to pass anything explicitly.

Reply:
Oh right, that was also based on not having the idtoken around but now I can wait for `useIsAuthorized()` so that works too